tests/test_data_collections.py

```diff
@@ -26,21 +18,39 @@ def test_derive() -> None:
     assert derived_definition.collection_type is None
 
 
-def test_compare() -> None:
+@pytest.mark.parametrize(
+    "definition_input, expected",
+    [
+        ({}, "DataCollectionDefinition(\n  is_timeless: False\n  has_cloud_coverage: False\n)"),
+        (
+            {"api_id": "X", "_name": "A"},
+            "DataCollectionDefinition(\n  api_id: X\n  is_timeless: False\n  has_cloud_coverage: False\n)",
+        ),
+        (
+            {"api_id": "Y", "is_timeless": True, "has_cloud_coverage": True},
+            "DataCollectionDefinition(\n  api_id: Y\n  is_timeless: True\n  has_cloud_coverage: True\n)",
+        ),
+    ],
+)
+def test_collection_repr(definition_input: Dict[str, Any], expected: str) -> None:
+    assert repr(DataCollectionDefinition(**definition_input)) == expected
+
+
+def test_collection_equality() -> None:
     def1 = DataCollectionDefinition(api_id="X", _name="A")
     def2 = DataCollectionDefinition(api_id="X", _name="B")
 
     assert def1 == def2
```

Comment:
are they the same object? (check with `is`)
how about definitions that should not be the same?
how about definitions that are the same but are not just one-parameter long?
Are these tests for collections or collection definitions?

Reply:
they should not be the same object just equal, more cases added

Reply:
would the tests look nicer if they were split into two? (one for equal, one for not-equal)

Reply:
Yes, done.